plugins/catalog-backend/src/search/DefaultCatalogCollator.ts

```diff
@@ -58,6 +58,11 @@ export class DefaultCatalogCollator implements DocumentCollator {
   async execute() {
     const baseUrl = await this.discovery.getBaseUrl('catalog');
     const res = await fetch(`${baseUrl}/entities`);
+
+    if (!res.ok) {
+      return [];
```

Comment:
So, the `.json()` still did throw before this fix, right? Just perhaps less cleanly. And in the current state, if you have a misconfiguration that e.g. gets an HTML login page back instead of json, then the line below would still throw. And the `fetch` itself can throw if there are network level issues.

Sounds to me like we might want to handle errors in the code that _calls_ this method instead. And to log!

Or perhaps in both. Maybe this method could have a try/catch around both fetch and json and all, and rethrow with a wrapped error with more explanatory text, even if that's sometimes not particularly graceful. Perhaps log and return the empty array instead, if that suits better.

Reply:
So the main issue is at startup if the collator runs before the catalog backend is running. There is a solution with a timeout, that waits with the startup, but that one is quite unstable. Thought about logging to, that would be a bigger change, but is an option. I will have a look later.

Reply:
In the first place I try to get rid of the `UnhandledPromiseRejectionWarning`, so error handling inside the `IndexBuilder` would be more suitable. I updated the PR to do error handling there, log on errors and skip indexing for the affected type. Other types are not affected and are still indexed, which I think is a resilient behavior if one of the sources is broken.

Reply:
Absolutely makes sense for now!  Thanks, @Fox32